### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by ejm714**
- [x] set minimum version for scikit-learn and use new `root_mean_squared_error` function ([API change details](https://scikit-learn.org/stable/whats_new/v1.4.html#sklearn-metrics))

> The squared parameter of [metrics.mean_squared_error](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_squared_error.html#sklearn.metrics.mean_squared_error) and [metrics.mean_squared_log_error](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_squared_log_error.html#sklearn.metrics.mean_squared_log_error) is deprecated and will be removed in 1.6. Use the new functions [metrics.root_mean_squared_error](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.root_mean_squared_error.html#sklearn.metrics.root_mean_squared_error) and [metrics.root_mean_squared_log_error](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.root_mean_squared_log_error.html#sklearn.metrics.root_mean_squared_log_error) instead

- [x] the SCL band has been reprocessed as it's picking up more water pixels and a couple more available images. this re-makes the tests assets in accordance with this ([reprocessing details](https://sentiwiki.copernicus.eu/web/s2-processing)). #155 tracks a full training to ensure consistency in data processing. EDIT: some of these differences are due to underlying variations in openCV implementations

> The improvements introduced in processing baselines up to baseline 04.00 (in operations since 25 January 2022) have been generalised to the historical archive:

> Improved L2A processing algorithms for scene classification and surface reflectance (PB 04.00 feature)

### Bonus fixes
- uncap matplotlib
- remove duplicate `clean` make command
- address pandas warnings

### CODE REVIEW

File: tests/test_experiment.py [hunk 1]
```diff
@@ -52,3 +53,8 @@ def test_cli_experiment(experiment_config_path):
         "results.json",
     ]:
         assert (Path(config.save_dir) / "metrics" / file).exists()
+
+    # re-write config to yaml directory without git hash to avoid changing test assets
```
[the remark this answers is not in the document]
Comment (klwetstone) [reply tests/test_experiment.py:57]: When I run `make assets`, I get a `config_artifact.yaml` that is written out with `last_commit_hash` included. `make assets` calls `cyfi/experiment.py`, so I don't think is affected by changes to `test_experiment.py`. 

Should `make assets` call `test_experiment.py` in some way instead of `experiment.py`?

File: tests/assets/experiment/config_artifact.yaml [hunk 2]
```diff
@@ -32,7 +32,6 @@ features_config:
   use_sentinel_bands:
   - B02
   - SCL
-last_commit_hash: ea440f57d459c98cf7cc03f7e24a0df05c5e23d2
```
Comment (klwetstone) [inline tests/assets/experiment/config_artifact.yaml:35]: When I run `make assets`, I get a `config_artifact.yaml` that is written out with `last_commit_hash` included.

`make assets` calls `cyfi/experiment.py`, so the hash is still written out when we re-create the assets even if re-running the tests doesn't write out the hash or change the `config_artifact.yaml` test asset. Is this expected? Or do we want to be able to rerun `make assets` without changing the asset files in addition to running `make tests` without changing the files?

File: tests/assets/experiment/sentinel_metadata_test.csv [hunk 3]
```diff
@@ -1,5 +1,5 @@
...
-d823c957bedf702bb1dd8d562e45a7b8,S2B_MSIL2A_20170713T173909_R098_T13TDE_20210210T083500,0.014554095195663475,66558.0,15,https://sentinel2l2a01.blob.core.windows.net/sentinel2-l2/13/T/DE/2017/07/13/S2B_MSIL2A_20170713T173909_N0212_R098_T13TDE_20210210T083500.SAFE/GRANULE/L2A_T13TDE_A001840_20170713T174714/IMG_DATA/R10m/T13TDE_20170713T173909_TCI_10m.tif
-f79c66af395e5de8fb3c5cc1b1ed6993,S2A_MSIL2A_20180521T154911_R054_T18TUK_20201012T122951,0.0155145746579417,6416.0,0,https://sentinel2l2a01.blob.core.windows.net/sentinel2-l2/18/T/UK/2018/05/21/S2A_MSIL2A_20180521T154911_N0212_R054_T18TUK_20201012T122951.SAFE/GRANULE/L2A_T18TUK_A015209_20180521T155356/IMG_DATA/R10m/T18TUK_20180521T154911_TCI_10m.tif
+494fb91d1fb8697e73b90e5d0c9420ed,S2A_MSIL2A_20190818T170851_R112_T15SUD_20201005T015156,0.010019129603060737,1096.0,8,https://sentinel2l2a01.blob.core.windows.net/sentinel2-l2/15/S/UD/2019/08/18/S2A_MSIL2A_20190818T170851_N0212_R112_T15SUD_20201005T015156.SAFE/GRANULE/L2A_T15SUD_A021702_20190818T171838/IMG_DATA/R10m/T15SUD_20190818T170851_TCI_10m.tif
```
Comment (klwetstone) [inline tests/assets/experiment/sentinel_metadata_test.csv:2]: When I rerun `make test` on this branch, I get different values for `num_water_pixels`. Mine generally align with the previous version of this file. I'm not sure why mine is different -- I started with a clean environment and re-installed the requirements.

Comment (klwetstone) [reply tests/assets/experiment/sentinel_metadata_test.csv:2]: I'm also getting slightly different values for `features_test.csv` and `features_train.csv`

**Review by klwetstone — COMMENTED** [review]
@ejm714 Generally looks good. I am getting slightly different values for the test assets when I rerun `make test` -- can we huddle quickly tomorrow?

Based on `sentinel_metadata_[train/test].csv`, I'm getting all the same tiles, but with different values for the number of water pixels. As a result of the different water masks, but features, r-squared values (results.json), and model are also different.

File: tests/assets/experiment/config_artifact.yaml [hunk 4]
```diff
@@ -5,7 +5,7 @@ cyfi_model_config:
   params:
     application: regression
     early_stopping_round: 100
-    feature_fraction: 1.0
```
[the remark this answers is not in the document]
Comment (ejm714) [reply tests/assets/experiment/config_artifact.yaml:8]: this is a bonus fix. we fixed the default value a while back but never did for the test assets: https://github.com/drivendataorg/cyfi/commit/ff314f9ad439701b2ccc0ffee1b0e4ad78b70775

File: pyproject.toml [hunk 5]
```diff
@@ -45,7 +45,7 @@ dependencies = [
   "pydantic>=2.0",
   "pystac",
   "pystac-client",
-  "scikit-learn",
```
[the remark this answers is not in the document]
Comment (ejm714) [reply pyproject.toml:48]: good call

File: tests/assets/experiment/sentinel_metadata_test.csv (hunk 3 above)
Comment (ejm714) [reply tests/assets/experiment/sentinel_metadata_test.csv:2]: hmmm that's very curious and good to know

File: tests/assets/experiment/config_artifact.yaml (hunk 2 above)
Comment (ejm714) [reply tests/assets/experiment/config_artifact.yaml:35]: great flag, i forgot we had `make assets`.

> Or do we want to be able to rerun make assets without changing the asset files in addition to running make tests without changing the files?

we do want this. i've put in a better fix that uses a temp dir as the save location for the test since we really shouldn't let tests make persistent changes

it's fine that `make assets` writes out the git hash since that lets us track provenance of when assets were created

File: tests/test_experiment.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ejm714) [reply tests/test_experiment.py:57]: see comment below

File: tests/test_experiment.py [hunk 6]
```diff
@@ -28,7 +29,13 @@ def test_experiment_config(train_data_path):
         )
 
 
-def test_cli_experiment(experiment_config_path):
+def test_cli_experiment(experiment_config_path, tmp_path):
+    # use tmp_path as the save location
+    config = ExperimentConfig.from_file(experiment_config_path)
+    config.save_dir = tmp_path
+    with (config.save_dir / "config_artifact.yaml").open("w") as fp:
```
Comment (klwetstone) [inline tests/test_experiment.py:36]: The to the experiment config also needs to be updated when calling the CLI

File: tests/assets/experiment/sentinel_metadata_test.csv (hunk 3 above)
Comment (ejm714) [reply tests/assets/experiment/sentinel_metadata_test.csv:2]: after a long investigation, the number of water pixel changes (with cascading effects on band summary stats) appear to be coming from OpeCV (due to variations in underlying hardware, operating systems, and library implementations). 

> Different processors and architectures may handle floating-point calculations with slight variations, leading to minor differences in the computed pixel values during interpolation.

this is the relevant part of the code https://github.com/drivendataorg/cyfi/blob/85c82cadc1f26286c3ab6c2238d682b1233da1a2/cyfi/data/features.py#L165-L169

File: pyproject.toml [hunk 7]
```diff
@@ -34,7 +34,7 @@ dependencies = [
   "gradio",
   "lightgbm",
   "loguru",
-  "matplotlib<3.8.0",
+  "matplotlib",
```
Comment (klwetstone) [inline pyproject.toml:37]: Why do we no longer need to enforce a version of matplotlib?

**Review by klwetstone — APPROVED** [review]
@ejm714 LGTM! Just one small question.

- All tests pass for me, and `make tests` doesn't change any of the test asset files
- `make assets` runs and does change test asset files (expected because of weird `opencv` behavior). The same images are selected for both train and test, although the number of water pixels is different

File: pyproject.toml (hunk 7 above)
Comment (ejm714) [reply pyproject.toml:37]: There was a mpl bug that was the cause of this ceiling (https://github.com/drivendataorg/cyfi/pull/104/commits/e379dec3599f705e192220de8740fc52286a6112) but I can't remember the specifics. In general, we only want floors not ceilings for packages. Ceilings are often a temporary workaround and it looked like this one was outdated (indeed as it was given tests pass with later versions)
